### PR TITLE
mini-compile refactor

### PR DESCRIPTION
**Pull request by shayne-fletcher**
refactor ghc-lib-test-mini-compile for readability. less compact but much easier to understand. a couple of minor incidental bugfixes.

### CODE REVIEW

**Comment by shayne-fletcher** [comment]
cc @garyverhaegen-da @akrmn @dylant-da @samuel-williams-da

**Review by dylant-da — APPROVED** [review]
Looks good to me, definitely more legible than piecewise changes everywhere

File: examples/ghc-lib-test-mini-compile/src/Main.hs [hunk 1]
```diff
@@ -54,175 +54,290 @@ main = do
...
         , pkgDatabase = Just []
-#endif
-#if defined (GHC_9_8) || defined (GHC_9_6) || defined (GHC_9_4) || defined (GHC_9_2)
-       -- Intentionally empty (temp file flags have moved to HscEnv).
-#else
         , dirsToClean = dirs_to_clean
         , filesToClean = files_to_clean
         , nextTempSuffix = next_temp_suffix
-#endif
-#if defined(DAML_UNIT_IDS)
```
[the remark this answers is not in the document]
Comment (shayne-fletcher) [reply examples/ghc-lib-test-mini-compile/src/Main.hs:108]: i searched daml's ghc and couldn't find reference to it. if it's the case that this is meaningless, could we have a patch that removes it from ghc-lib please @dylant-da?

**Pull request merged**